src/main/java/hudson/tasks/Mailer.java

```diff
@@ -400,24 +408,22 @@ private static Session createSession(String smtpHost, String smtpPort, boolean u
                 props.put("mail.smtp.starttls.enable", "true");
                 props.put("mail.smtp.starttls.required", "true");
             }
-            if(smtpAuthUserName!=null)
-                props.put("mail.smtp.auth","true");
+
+            Optional<Authenticator> authenticator = Optional.ofNullable(credentialsId)
+                    .map(id -> CredentialsMatchers.firstOrNull(
+                            CredentialsProvider.lookupCredentials(StandardUsernamePasswordCredentials.class, (Item) null, ACL.SYSTEM, Collections.emptyList()),
```

Comment:
Do I need to do anything extra for credentials tracking here, or does a call to `lookupCredentials` append to the tracking log automatically?